SprawdzanieNIPu/ConsoleApplication1/NIP.cs

```diff
@@ -1,72 +1,111 @@
-﻿using System;
+﻿//-----------------------------------------------------------------------
+// <copyright file="NIP.cs" company="CompanyName">
+// Company copyright tag.
+// </copyright>
+//----------------------------------------------------------------------- 
 
 namespace ConsoleApplication1
-{
-    class NIP
-    {
-        public string numer;
+ {
+    using System;
 
-        public NIP(string nr)
-        {
-            this.numer = nr.Trim();
-        }
+    /// <summary>
+    /// Klasa służy do sprawdzenia, czy wprowadzony ciąg znaków jest poprawnym numerem NIP.
+    /// </summary>
+ internal class NIP
+ {
+    /// <summary>
+    /// Wprowadzony numer po usunięciu białych znaków
+    /// </summary>
+   private string numer;
 
-        public Boolean dlugoscOK
-        {
-            get
-            {
-                return this.numer.Length == 10;
-            }
-        }
+    /// <summary>
+    /// Initializes a new instance of the NIP class.
+    /// </summary>
+    /// <param name="nr">Wprowadzony ciąg znaków</param>
+   public NIP(string nr)
+   {
+    this.numer = nr.Trim();
+   }
+   
+    /// <summary>
+    /// Gets a value indicating whether długość jest poprawna
+    /// </summary>
+   public bool DlugoscOK
+   {
+    get
+    {
+     return this.numer.Length == 10;
+    }
+   }
+   
+    /// <summary>
+    /// Gets - sumę kontrolną
+    /// </summary>
+   public int Kontrolna
+   {
+    get
+    {
+     int[] wagi = new int[] { 6, 5, 7, 2, 3, 4, 5, 6, 7 };
+     double x = 0;
+     double liczba = 0;
 
-        public int kontrolna
-        {
-            // zwraca sumę kontrolną
-            get
-            {
-                int[] wagi = new int[] {6, 5, 7, 2, 3, 4, 5, 6, 7};
-                double x=0;
-                double liczba=0;
+     for (int i = 0; i < wagi.Length; i++) 
+     {
+      x = Convert.ToDouble(this.numer.Substring(i, 1));
```

Comment:
@naleczynskia Miałem problem z konwersją na Int. To jest oczywiście do poprawienia.